recipes/migrate.rb

```diff
@@ -0,0 +1,73 @@
+versions = node['onlinefs']['migrate']['versions'].split(/\s*,\s*/)
```

Comment:
Let's not add a new recipe that should be run only during upgrades, otherwise people will forget to include it. Run the migration script as part of the default recipe and make sure it's running only if it's an upgrade from before 3.5.0

You can see an example here:
https://github.com/logicalclocks/kagent-chef/blob/26c66c4873bf34ed16236540ec16b96fb411c8cd/providers/hopsify.rb#L84

Reply:
I am not sure about the procedure of upgrading onlinefs. If it is done by first install onlinefs and then restore the mysql database, then the migration code cannot be in default recipe.

Reply:
The order of execution is defined here: https://github.com/logicalclocks/onlinefs-chef/blob/master/Karamelfile - There is a transitive dependency, i.e. to have Hopsworks running we need to have the database running.

Everything in onlinefs-chef is done in the default recipe. I would run the migration first and then update the onlinefs docker images and restart the services.